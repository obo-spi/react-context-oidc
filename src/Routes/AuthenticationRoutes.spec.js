import * as React from 'react';
import { render } from 'react-testing-library';
import Component from './AuthenticationRoutes';

jest.mock('../OidcComponents', () => ({
  NotAuthenticated: 'NotAuthenticated',
  NotAuthorized: 'NotAuthorized',
}));
jest.mock('../Callback', () => ({
  Callback: 'Callback',
  SilentCallback: 'SilentCallback',
}));
jest.mock('react-router', () => ({
  Route: 'Route',
  Switch: 'Switch',
}));

describe('Authenticating test suite', () => {
  it('renders correctly', () => {
    const matchMock = {
      url: 'http://url.com',
    };
    const ComponentRend = Component();
    const { asFragment } = render(<ComponentRend match={matchMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
