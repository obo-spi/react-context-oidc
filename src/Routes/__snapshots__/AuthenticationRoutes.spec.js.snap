// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticating test suite renders correctly 1`] = `
<DocumentFragment>
  <switch>
    <route
      component="Callback"
      path="http://url.com/callback"
    />
    <route
      component="SilentCallback"
      path="http://url.com/silent_callback"
    />
    <route
      component="NotAuthenticated"
      path="http://url.com/not-authenticated"
    />
    <route
      component="NotAuthorized"
      path="http://url.com/not-authorized"
    />
  </switch>
</DocumentFragment>
`;
